File: country.php
<?php
header('Content-Type: application/json');
header('Access-Control-Allow-Origin: *');
header('Access-Control-Allow-Methods: GET, POST, OPTIONS');
header('Access-Control-Allow-Headers: Content-Type');

$host = 'localhost';
$user = 'root';
$pass = '';
$db = 'world_db';

$conn = new mysqli($host, $user, $pass, $db);

if ($conn->connect_error) {
    echo json_encode(['error' => 'Connection failed']);
    exit;
}

$code = isset($_GET['code']) ? $_GET['code'] : '';

if (empty($code)) {
    echo json_encode(['error' => 'Country code required']);
    exit;
}

$sql = "SELECT id, name, iso2, iso3, capital, population, currency, emoji 
        FROM countries WHERE id = '$code' OR iso2 = '$code'";

$result = $conn->query($sql);

if ($result && $result->num_rows > 0) {
    $country = $result->fetch_assoc();
    echo json_encode($country, JSON_UNESCAPED_UNICODE);
} else {
    echo json_encode(['error' => 'Country not found']);
}

$conn->close();
?>

File: lib/models/country.dart
class Country {
  final int id;
  final String name;
  final String code;
  final String iso3;
  final String capital;
  final int population;
  final String currency;
  final String emoji;
  final double? area;
  final String? languages;
  final String? phoneCode;
  final String? timezones;
  final String? region;
  final String? subregion;
  final String? nativeName;
  final String? nationality;

  Country({
    required this.id,
    required this.name,
    required this.code,
    required this.iso3,
    required this.capital,
    required this.population,
    required this.currency,
    required this.emoji,
    this.area,
    this.languages,
    this.phoneCode,
    this.timezones,
    this.region,
    this.subregion,
    this.nativeName,
    this.nationality,
  });

  factory Country.fromJson(Map<String, dynamic> json) {
    return Country(
      id: int.tryParse(json['id']?.toString() ?? '0') ?? 0,
      name: json['name'] ?? '',
      code: json['iso2'] ?? json['code'] ?? '',
      iso3: json['iso3'] ?? '',
      capital: json['capital'] ?? '',
      population: int.tryParse(json['population']?.toString() ?? '0') ?? 0,
      currency: json['currency'] ?? '',
      emoji: json['emoji'] ?? '🌍',
      area: json['area_sq_km'] != null
          ? double.tryParse(json['area_sq_km'].toString())
          : null,
      languages: json['languages'],
      phoneCode: json['phonecode'],
      timezones: json['timezones'],
      region: json['region'],
      subregion: json['subregion'],
      nativeName: json['native'],
      nationality: json['nationality'],
    );
  }
}
